src/accessors-geog.cpp

```diff
@@ -0,0 +1,72 @@
+#include <s2geography.h>
+
+#include "geography.hpp"
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+PyObjectGeography centroid(PyObjectGeography a) {
+    const auto& a_ptr = a.as_geog_ptr()->geog();
+    auto s2_point = s2geog::s2_centroid(a_ptr);
+    std::unique_ptr<Point> point =
+        make_geography<s2geog::PointGeography, spherely::Point>(s2_point);
+    return PyObjectGeography::from_geog(std::move(point));
+}
+
+PyObjectGeography boundary(PyObjectGeography a) {
+    const auto& a_ptr = a.as_geog_ptr()->geog();
+    auto s2_obj = s2geog::s2_boundary(a_ptr);
+    // TODO return specific subclass
+    auto geog_ptr = std::make_unique<spherely::Geography>(std::move(s2_obj));
```

Comment:
Boundary can return various geography types (empty collection for point, multipoint for linestring, linestring for polygon), so for now using the generic `Geography`, which will also give this parent class at the python level. We will have to add some utility to infer the type and cast to correct subclass.